File: 2645-pass-the-pillow/pass-the-pillow.js
/**
 * @param {number} n
 * @param {number} time
 * @return {number}
 */
var passThePillow = function (n, time) {
    let r = Math.floor(time % (n - 1));
    let q = Math.floor(time / (n - 1));

    if (q % 2 === 0) {
        return (r + 1);
    }
    else {
        return (n - r);
    }
};

File: 2645-pass-the-pillow/pass-the-pillow.py
class Solution:
    def passThePillow(self, n: int, time: int) -> int:
        r = time % (n - 1)
        q = time // (n - 1)

        if (q % 2 == 0):
            return (r + 1)
        else:
            return (n - r)

        
        